### PR TITLE
Scheduler docs

### LINKED ISSUE #682: Create docpage about how to use async scheduler
There are various options and flags that are pretty handy in non-trivial use.  Currently the only documentation is more research-y.  It would be good to have a doc page that was about “How to run the thing you just made” (feedback from @seibert)

### PR DESCRIPTION
**Pull request by jcrist**
- Add doc page giving overview of scheduler interface, how to use `get` and `compute` functions, and how to configure the schedulers
- Can now build docs locally with readthedocs template, which helps with debugging the format
- Added headers to the sidebar to break pages into sections (example sidebar [here](https://read-the-docs.readthedocs.org/en/latest/))

Closes #682.

### CODE REVIEW

File: docs/source/scheduler-overview.rst [hunk 1]
```diff
@@ -0,0 +1,146 @@
+Scheduler Overview
+==================
+
+Once a dask graph is created, a scheduler is needed to run it. Dask currently
```
Comment (mrocklin) [inline docs/source/scheduler-overview.rst]: Consider active voice:

    After we create a dask graph we use a scheduler to run it.

Or

File: docs/source/scheduler-overview.rst [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    >>> import dask.array as da
+    >>> x = da.arange(100, chunks=10)
+    >>> x.sum().compute()
+    4950
+
+The compute method takes a number of keywords:
+
+- ``get``: a scheduler ``get`` function, overrides the default for the collection
+- ``**kwargs``: extra keywords to pass on to the scheduler ``get`` function (see :ref:`configuring-schedulers`)
+
```
Comment (mrocklin) [inline docs/source/scheduler-overview.rst]: So you can combine any compute function with any get function.  E.g.

    >>> x.sum().compute(get=dask.threaded.get)  # use threads
    >>> x.sum().compute(get=dask.async.get_sync)  # use single thread

File: docs/source/scheduler-overview.rst [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+``compute`` function that takes multiple collections and returns multiple
+results. This merges the graphs from each collection, so intermediate results
+are shared.
+
+.. code-block:: python
+
+    >>> y = (x + 1).sum()
+    >>> z = (x + 1).mean()
+    >>> da.compute(y, z)    # Compute y and z, sharing intermediate results
+    (5050, 50.5)
```
Comment (mrocklin) [inline docs/source/scheduler-overview.rst]: This point might benefit from repetition.  

Perhaps expand on why this is good.  Particularly stating that we will only have to compute `x + 1` once.

**Comment by mrocklin** [comment]
This looks good.  I added some notes on a first pass but these could be ignored.  

If possible it would be good to emphasize the use of `dask.async.get_sync` for debugging.  This is a frequently asked question.

**Comment by jcrist** [comment]
Addressed the points I agreed with, biggest change is an addition of a debugging section. This could use another review.

**Pull request merged**